pkg/cli/admin/ocpcertificates/regeneratemco/command.go

```diff
@@ -177,5 +187,78 @@ func run(o *RegenerateMCOOptions, args []string) error {
 		return err
 	}
 
+	if !o.modifyUserData {
+		return nil
+	}
+
+	// Get the updated CA cert
+	mcoSecrets := clientset.CoreV1().Secrets(mcoNamespace)
+	mcsSecret, err := mcoSecrets.Get(ctx, newMCSCASecret, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("cannot read MCS secret: %w", err)
+	}
+
+	caCert, exists := mcsSecret.Data[CASecretCertName]
+	if !exists {
+		return fmt.Errorf("could not find MCS CA cert at %s after rotation", CASecretCertName)
+	}
+	encodedCACert := base64.StdEncoding.EncodeToString(caCert)
+
+	// Get user-data-secret
+	mapiSecrets := clientset.CoreV1().Secrets(mapiNamespace)
+	secretList, err := mapiSecrets.List(ctx, metav1.ListOptions{})
```

Comment:
So I think this is fine, but we will be trying to edit the `user-data-managed` secrets generated/owned by the MCO today that are unused.  I think the MCO will edit them back?

In the general case as far as I understand it, probably what we should be doing is iterating over the *machineset* objects, and then we can find references to their user data from there.

Reply:
Right, to rotate those, we'd have to change the MCO code I think. Let's check if we would need to do so or not